.github/workflows/main.yml

```diff
@@ -108,3 +107,90 @@ jobs:
         run: ./scripts/release_canary.sh
         env:
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Preparing tarballs of packages for testing templates installation
+        run: node ./scripts/build-tarballs.js
+
+      # TODO: in the future we want to extract this step into a separate job to be able
+      # to test this in multiple environments (macOS, Window) and with different node
+      # versions (10, 12). This can be done using the action strategy matrix.
+      # However, in order to do that, we need to be able to persist the tarballs between
+      # jobs. This can be done by using the upload/download artifacts actions.
+      # However, at this time there is no option to clean up the artifacts automatically
+      # after the workflow ends, and the artifacts will stay for 90 days.
+      # Some related issues:
+      # - https://github.com/actions/upload-artifact/issues/5
+      # - https://github.com/actions/upload-artifact/issues/45
+      - name: Installing and building template starter
+        run: node ./scripts/install-template.js
+        env:
+          TEMPLATE_NAME: starter
+
+  #     - name: Uploading tarball artifacts
```

Comment:
From here below is the "ideal setup". I kept it commented for future references.